src/superqt/collapsible/_collapsible.py

```diff
@@ -132,11 +132,10 @@ def _expand_collapse(
             self._animation.setEndValue(_content_height)
             self._is_animating = True
             self._animation.start()
+            self.toggled.emit(direction == QPropertyAnimation.Direction.Forward)
         else:
             self._content.setMaximumHeight(_content_height if forward else 0)
 
-        self.toggled.emit(direction == QPropertyAnimation.Direction.Forward)
```

Comment:
But if animate is  `False` then it would not be emitted at all.

Do you know why it was being called three times before @ppwadhwa or @tlambert03 ?

Reply:
Ok what if have something like this, that way we never emit signals when calling expand_collapse from within the eventFilter
```python
...

    def _expand_collapse(
        self, direction: QPropertyAnimation.Direction, animate: bool = True, emit: bool = True
    ):
        if self._locked:
            return

        forward = direction == QPropertyAnimation.Direction.Forward
        text = self._EXPANDED if forward else self._COLLAPSED

        self._toggle_btn.setChecked(forward)
        self._toggle_btn.setText(text + self._toggle_btn.text()[len(self._EXPANDED) :])

        _content_height = self._content.sizeHint().height() + 10
        if animate:
            self._animation.setDirection(direction)
            self._animation.setEndValue(_content_height)
            self._is_animating = True
            self._animation.start()
        else:
            self._content.setMaximumHeight(_content_height if forward else 0)

        if emit:
            self.toggled.emit(direction == QPropertyAnimation.Direction.Forward)

    def _toggle(self):
        self.expand() if self.isExpanded() else self.collapse()

    def eventFilter(self, a0: QObject, a1: QEvent) -> bool:
        """If a child widget resizes, we need to update our expanded height."""
        if (
            a1.type() == QEvent.Type.Resize
            and self.isExpanded()
            and not self._is_animating
        ):
            # Since this can be called several times when child widgets resize, we need to
            # ensure we do not emit the signal while we are animating
            self._expand_collapse(QPropertyAnimation.Direction.Forward, animate=False, emit=False)
        return False

...

```